app/animals/otherAnimalController.ts

```diff
@@ -0,0 +1,43 @@
+﻿namespace dogsrus.virtdog {
+  export class OtherAnimalController {
+    public otherAnimals: IAnimal[] = [];
+    public selectedAnimal: IAnimal;
+    public lastAction: string = '';
+
+    static $inject = ['$rootScope', 'eventNames'];
+    constructor(private $rootScope: ng.IRootScopeService, private eventNames: EventNames) {
+      this.intializeAnimalList(eventNames);
+    }
+
+    public performAction() {
+      // in future may keep track of last action so we can respond back and forth
+      this.lastAction = (<any>this.selectedAnimal).defaultAction;
```

Comment:
Do you really need an any cast here? The animal interface does have a default action...